gcloud-java-examples/src/main/java/com/google/gcloud/examples/DnsExample.java

```diff
@@ -0,0 +1,516 @@
+    }
+
+  }
+
+  private static class DeleteDnsRecordAction implements DnsAction {
+
+    /**
+     * Deletes a DNS record of type A from the given zone. The last parameter is ttl and it is not
+     * required.
+     */
+    @Override
+    public void run(Dns dns, String... args) {
+      String zoneName = args[0];
+      String recordName = args[1];
+      String ip = args[2];
+      DnsRecord record = DnsRecord.builder(recordName, DnsRecord.Type.A)
+          .records(ImmutableList.of(ip))
+          .build();
+      if (args.length > 3) {
+        Integer ttl = Integer.valueOf(args[3]);
+        record = record.toBuilder().ttl(ttl, TimeUnit.SECONDS).build();
+      }
+      ChangeRequest changeRequest = ChangeRequest.builder()
+          .delete(record)
+          .build();
+      changeRequest = dns.applyChangeRequest(zoneName, changeRequest);
+      System.out.printf("The request for deleting A record %s for zone %s was successfully " +
+          "submitted and assigned ID %s.%n", recordName, zoneName, changeRequest.id());
+      while (changeRequest.status().equals(ChangeRequest.Status.PENDING)) {
+        try {
+          Thread.sleep(500);
+        } catch (InterruptedException e) {
+          System.err.println("Thread was interrupted while waiting.");
+        }
+        changeRequest = dns.getChangeRequest(zoneName, changeRequest.id());
+      }
+      System.out.printf("The deletion has been completed.%n");
+    }
+
+    @Override
+    public String params() {
+      return "<zone_name> <record_name> <ip_address> [<ttl>]";
+    }
+
+    @Override
+    public boolean check(String... args) {
+      if (args.length == 4) {
+        try {
+          Integer.valueOf(args[3]);
+        } catch (Exception ex) {
+          throw new IllegalArgumentException(ex);
+        }
+        return true;
+      } else {
+        return args.length == 3;
+      }
+    }
+  }
+
+  private static class AddDnsRecordAction implements DnsAction {
+
+    /**
+     * Adds a DNS record of type A. The last parameter is ttl and is not required.
+     */
+    @Override
+    public void run(Dns dns, String... args) {
+      String zoneName = args[0];
+      String recordName = args[1];
+      String ip = args[2];
+      DnsRecord record = DnsRecord.builder(recordName, DnsRecord.Type.A)
+          .records(ImmutableList.of(ip))
+          .build();
+      if (args.length > 3) {
+        Integer ttl = Integer.valueOf(args[3]);
+        record = record.toBuilder().ttl(ttl, TimeUnit.SECONDS).build();
+      }
+      ChangeRequest changeRequest = ChangeRequest.builder()
+          .add(record)
+          .build();
+      changeRequest = dns.applyChangeRequest(zoneName, changeRequest);
+      System.out.printf("The request for adding A record %s for zone %s was successfully " +
+          "submitted and assigned ID %s.%n", recordName, zoneName, changeRequest.id());
+      while (changeRequest.status().equals(ChangeRequest.Status.PENDING)) {
```

Comment:
Same as above, if change requests usually take a while to come through, print something.